### DISCUSSION TITLE
tb-profiler collate :   what is the difference between  --full   vs --all_variants vs default

### DISCUSSION THREAD

**Discussion by harrismia** [jodyphelan/TBProfiler#146] [Q&A] [answered]
Hello, 

    For the collate feature, what is the difference between the different options? :    

**(default)** 

    vs 

**--full**                Output mutations in main result file (default: False)

  vs 

**--all_variants**        Output all variants in variant matrix (default: False)
.

**Comment by jodyphelan** [comment, marked as answer]
Hi Michael,

The `--full` argument is a relic from the past. The collated report (tbprofiler.txt) used to only show "R" when a mutation was detected unless the `--full` argument was provided, in which case it would list the mutation instead. The second case is now the default option so this argument is redundant. It is still in there for compatibility between versions, however at some point I will remove it.

The same goes for the `--all_variants`, the default behaviour used to be that the tbprofiler.variants.txt file only contained the drug resistance variants found in the sample, however it now lists all variants found in the drug resistance database (which used to happen with `--all_variants`). This function could actually do with some improvement, as it might miss some cases (such as novel frameshifts). I'll have a look at this asap.

Let me know if these defaults make sense for your purpose or whether you would suggest some changes.

Jody

**Reply by harrismia** [reply]
Hi Jody, 

Thanks very much for the information.  Sorry for the delay in responding.  What you described for the default parameters sounds reasonable to me.    

One thing that may be useful is to be able to differentiate between the drug resistance confidence levels: "High", "Intermediate", "Low", "Indeterminate" SNPs  in the SNP database.  This may be through command line arguments to specify for example that only. "High" and "Intermediate" SNPs should be reported. Also,  if all confidence levels are used then perhaps the "High" could be marked somehow to differentiate them from the lower confidence SNPs.

Another feature that would be nice is the ability to create a full genome VCF.  We currently create these using our own pipeline but it would be nice to  have the ability to create these with TB Profiler.  

Thanks for making this fantastic tool available to the research community!

Best, 

   Michael

**Reply by jodyphelan** [reply]
No problem, good to know they are reasonable defaults.

> One thing that may be useful is to be able to differentiate between the drug resistance confidence levels

That sounds like a great idea. This information is actually stored internally, so it hopefully shouldn't take too much work to build in filtering based on the confidence level. 

> Another feature that would be nice is the ability to create a full genome VCF

This has also been requested by other users so I will bump it up the priority list!

Thanks for all the continued feedback!

Jody
